components/org.wso2.carbon.crypto.provider/src/test/java/org/wso2/carbon/crypto/provider/SymmetricKeyInternalCryptoProviderTest.java

```diff
@@ -0,0 +1,128 @@
+
+    SymmetricKeyInternalCryptoProvider symmetricKeyInternalCryptoProvider;
+    private static final String OLD_SECRET = "f7b2b39207523b43a56540d30656b19b";
+    private static final String SECRET = "22b97077751bc066067ffeefb83a1c16";
+    private static final String PLAIN_TEXT = "wso2carbon";
+    private static final String JCE_PROVIDER = "SunJCE";
+
+    private static final String AES_ALGORITHM = "AES";
+    private static final String AES_GCM_ALGORITHM = "AES/GCM/NoPadding";
+    private static final String CIPHER_TEXT = "nQ2r9QHRYJP+i88JDsyOpA==";
+    private static final String CIPHER_TEXT_FOR_OLD_SECRET = "ueAbJkt7K+s20lIuvi9/1A==";
+
+    @BeforeClass
+    public void init() throws Exception {
+
+        symmetricKeyInternalCryptoProvider = new SymmetricKeyInternalCryptoProvider(SECRET);
+    }
+
+    @Test(description = "Test encryption using the default secret.")
+    public void testEncrypt() throws CryptoException {
+
+        byte[] plainTextBytes = PLAIN_TEXT.getBytes();
+        byte[] encryptedText = symmetricKeyInternalCryptoProvider.encrypt(
+                plainTextBytes, AES_ALGORITHM, JCE_PROVIDER);
+        assertEquals(new String(Base64.getEncoder().encode(encryptedText)), CIPHER_TEXT);
+    }
+
+    @Test(description = "Test encryption using a custom secret.")
+    public void testEncryptUsingCustomSecret() throws CryptoException {
+
+        byte[] plainTextBytes = PLAIN_TEXT.getBytes();
+        byte[] encryptedText = symmetricKeyInternalCryptoProvider.encrypt(
+                plainTextBytes, AES_ALGORITHM, JCE_PROVIDER, false, OLD_SECRET);
+        // TODO: should return CIPHER_TEXT_FOR_OLD_SECRET
+        assertEquals(new String(Base64.getEncoder().encode(encryptedText)), CIPHER_TEXT);
+    }
+
+    @Test(description = "Test decryption using the default secret.")
+    public void testDecrypt() throws CryptoException {
+
+        byte[] encryptedText = Base64.getDecoder().decode(CIPHER_TEXT);
+        byte[] decryptedText = symmetricKeyInternalCryptoProvider.decrypt(
+                encryptedText, AES_ALGORITHM, JCE_PROVIDER);
+        assertEquals(new String(decryptedText), PLAIN_TEXT);
+    }
+
+    @Test(description = "Test decryption using a custom secret.")
+    public void testDecryptUsingCustomSecret() throws CryptoException {
+
+        byte[] encryptedText = Base64.getDecoder().decode(CIPHER_TEXT_FOR_OLD_SECRET);
+        byte[] decryptedText = symmetricKeyInternalCryptoProvider.decrypt(
+                encryptedText, AES_ALGORITHM, JCE_PROVIDER, OLD_SECRET);
+        assertEquals(new String(decryptedText), PLAIN_TEXT);
+    }
+
+    @Test(description = "Test decryption using an invalid custom secret.")
+    public void testDecryptUsingInvalidCustomSecret() throws CryptoException {
+
+        byte[] encryptedText = Base64.getDecoder().decode(CIPHER_TEXT);
+        byte[] decryptedText = symmetricKeyInternalCryptoProvider.decrypt(
+                encryptedText, AES_ALGORITHM, JCE_PROVIDER, OLD_SECRET);
```

Comment:
Decryption with old secret works but encryption doesn't?

Reply:
Seems like the variable name is misleading. Will update